src/components/EditRecipe.tsx

```diff
@@ -0,0 +1,655 @@
+import React, { useState, useEffect } from "react";
+import {
+  Box,
+  Container,
+  IconButton,
+  Image,
+  Input,
+  InputGroup,
+  InputRightElement,
+  FormControl,
+  FormHelperText,
+  FormLabel,
+  Flex,
+  Grid,
+  GridItem,
+  Heading,
+  Select,
+  Text,
+  Textarea,
+  useToast
+} from "@chakra-ui/react";
+import { AddIcon, CheckIcon, CloseIcon, MinusIcon } from "@chakra-ui/icons";
+import { useParams, useNavigate } from "react-router-dom";
+import { getSingleRecipe, editSingleRecipe } from "../utils/fetchData";
+import { SavedRecipe, SavedIngredient, RecipeTag } from "../utils/types";
+import IngredientAmountHandle from "./IngredientAmountHandle";
+
+const EditRecipe = () => {
+  const [recipe, setRecipe] = useState<SavedRecipe | null>(null);
+  const [ingredients, setIngredients] = useState<Array<SavedIngredient> | null>(
+    null
+  );
+  const [tags, setTags] = useState<Array<RecipeTag> | null>(null);
+  const [diets, setDiets] = useState<Array<string> | null>(null);
+  const { slug } = useParams();
+  const recipeId = slug;
+  const navigate = useNavigate();
+  const toast = useToast();
+
+  useEffect(() => {
+    if (recipeId === undefined) return;
+    getSingleRecipe(recipeId)
+      .then(response => {
+        setRecipe(response.data);
+        setIngredients(
+          response.data.recipeIngredients.map(
+            ({ _id, ...rest }: SavedIngredient) => {
+              return rest;
+            }
+          )
+        );
+        setTags(
+          response.data.recipeTags.map((tag: RecipeTag) => {
+            return { tagName: tag.tagName };
+          })
+        );
+        setDiets(response.data.recipeSpecialDiets);
+      })
+      .catch(error => {
+        console.log(error);
+      });
+  }, [recipeId]);
+
+  const saveRecipe = () => {
+    if (recipeId === undefined) return;
+    if (recipe === null) return;
+    editSingleRecipe(recipeId, recipe)
+      .then(response => {
+        console.log(response);
+        navigate(`/saved-recipes/${recipeId}`);
+      })
+      .catch(error => {
+        console.log(error);
+        toast({
+          title: "Error",
+          description: "Server wasn't be able to edit your recipe",
+          status: "error",
+          duration: 2000,
+          isClosable: true,
+          position: "top"
+        });
+      });
+  };
+
+  if (recipe === null) return null;
+  if (ingredients === null) return null;
+  if (tags === null) return null;
+  if (diets === null) return null;
```

Comment:
What is the purpose of these lines?

Reply:
Usually, all of your return statements for a functional component should be at the bottom to make it easier to read.